gnovm/pkg/gnolang/static_analysis_test.go

```diff
@@ -0,0 +1,351 @@
+package gnolang
+
+import "testing"
+
+func TestStaticAnalysisShouldPanic(t *testing.T) {
+	defer func() {
+		if r := recover(); r == nil {
+			t.Errorf("The code did not panic")
+		}
+	}()
+
+	cases := []string{
+		`package test
+				func main() {
+					invalidSwitch()
+				}
+			func invalidSwitch() int {
+			mockVar := map[string]int{
+				"apple":      10,
+				"banana":     20,
+				"orange":     30,
+				"mango":      5,
+				"watermelon": 15,
+			}
+		
+			for k, v := range mockVar {
+				switch k {
+				case "apple":
+					if v < 50 {
+						return(v)
+					} else {
+						return(50)
+					}
+				case "banana":
+					if v > 10 {
+						return(v)
+					} else {
+						return(10)
+					}
+				case "orange":
+					if v > 30 {
+					} else {
+						return(30)
+					}
+				case "mango":
+					if v == 5 {
+						return(v)
+					} else {
+						return(5)
+					}
+				case "watermelon":
+					if v == 15 {
+						return(v)
+					} else {
+						return(15)
+					}
+				default:
+					return 0
+				}
+			}
+		}
+		`,
+		`package test
+				func main() {
+		
+				}
+		
+			func invalidLabel() int{
+				Outer:
+					for {
+						for {
+							break Outer
+						}
+					}
+			}
+		`,
+		`package test
+				func main() {
+					oddOrEven(3)
+				}
+		
+			func oddOrEven(x int) bool {
+				for i := 0; i < x; i++ {
+					if(x % 2 == 0){
+		
+					} else {
+						return false
+					}
+				}
+			}
+			`,
+		`package test
+				func main() {
+					sumArr([]int{1,1,1})
+				}
+		
+			func sumArr(x []int) int {
+				sum := 0
+				for _, s := range x {
+					sum = sum + s
+				}
+			}
+		`,
+		`package test
+				func main() {
+					invalidSwitch(3)
+				}
+		
+			func invalidSwitch(x int) int {
+				switch x {
+					case 1:
+						return 1
+					case 2:
+						return 2
+				}
+			}
+		`,
+		`package test
+				func main() {
+					invalidCompareBool(4,5)
+				}
+		
+				func invalidCompareBool(a, b int) bool {
+					if a > b {
+						return true
+					} else {
+					}
+				}
+				`,
+		`package test
+				func main() {
+					invalidIfStatement(6)
+				}
+		
+		func invalidIfStatement(x int) int {
+		   if x > 5 {
+		
+		   } else {
+		       return x
+		   }
+		}
+		
+		`,
+	}
+
+	for _, s := range cases {
+		m := NewMachine("test", nil)
+
+		n := MustParseFile("main.go", s)
+		m.RunFiles(n)
+		m.RunMain()
+	}
+}
+
+func TestStaticAnalysisShouldPass(t *testing.T) {
+	cases := []string{
```

Comment:
These test cases should be organizes as a test table that has a name, so we can execute individual test cases and see which one fails / succeeds